control-plane/tests/tests/nexus.rs

```diff
@@ -4,22 +4,198 @@ pub mod common;
+        .with_rest_timeout(std::time::Duration::from_secs(1))
+        // don't log whilst we have the allow_fail
+        .compose_build(|c| c.with_logs(false))
+        .await
+        .unwrap();
+
+    for size_mb in &vec![6, 10, 100] {
+        let size = size_mb * 1024 * 1024;
+        let disk = || format!("malloc:///disk?size_mb={}", size_mb);
+        let sizes = vec![Ok(size / 2), Ok(size), Err(size + 512)];
+        for test in sizes {
+            let size = result_either!(test);
+            test_result(&test, async {
+                let nexus = cluster
+                    .rest_v0()
+                    .create_nexus(v0::CreateNexus {
+                        node: cluster.node(0),
+                        uuid: v0::NexusId::new(),
+                        size,
+                        children: vec![disk().into()],
+                    })
+                    .await;
+                if let Ok(nexus) = &nexus {
+                    cluster
+                        .rest_v0()
+                        .destroy_nexus(v0::DestroyNexus {
+                            node: nexus.node.clone(),
+                            uuid: nexus.uuid.clone(),
+                        })
+                        .await
+                        .unwrap();
+                }
+                nexus
+            })
+            .await
+            .unwrap();
+        }
+    }
+
+    for size_mb in &vec![1, 2, 4] {
+        let size = size_mb * 1024 * 1024;
+        let disk = || format!("malloc:///disk?size_mb={}", size_mb);
+        let sizes = vec![Err(size / 2), Err(size), Err(size + 512)];
+        for test in sizes {
+            let size = result_either!(test);
+            test_result(&test, async {
+                let nexus = cluster
+                    .rest_v0()
+                    .create_nexus(v0::CreateNexus {
+                        node: cluster.node(0),
+                        uuid: v0::NexusId::new(),
+                        size,
+                        children: vec![disk().into()],
+                    })
+                    .await;
+                if let Ok(nexus) = &nexus {
+                    cluster
+                        .rest_v0()
+                        .destroy_nexus(v0::DestroyNexus {
+                            node: nexus.node.clone(),
+                            uuid: nexus.uuid.clone(),
+                        })
+                        .await
+                        .unwrap();
+                }
+                nexus
+            })
+            .await
+            .unwrap();
+        }
+    }
+}
+
+#[actix_rt::test]
+async fn create_nexus_local_replica() {
+    let size = 10 * 1024 * 1024;
     let cluster = ClusterBuilder::builder()
         .with_pools(1)
-        .with_replicas(2, 5 * 1024 * 1024, v0::Protocol::Off)
+        .with_replicas(1, size, v0::Protocol::Off)
         .build()
         .await
         .unwrap();
 
+    let replica = format!("bdev:///{}", Cluster::replica(0, 0));
```

Comment:
`bdev://` is being kept around for backwards compatibility, new code should use `loopback://`.

Reply:
Ah thanks, should probably also change the volume agent - I'm changing the agents for another pr so I'll do it there.